### ISSUE TITLE
Whose state should the environment return?

### ISSUE THREAD

**Issue by wessle** [DavidNKraemer/SoCG21#5]
### Standard RL setting
A typical `gym` environment `env`, when `action` is taken in the current `state`, returns a tuple of the form
```
next_state, reward, done, info = env.step(action)
```
where `reward` depends on `state, action`. For one step of Q-learning to be carried out, we need access to the tuple
```
state, action, reward, next_state, done
```
which is appended to a replay buffer that is used in training.

### Issue in our RL setting
In our situation there are multiple agents acting in sequence, yet we are representing the environment as a single `gym` object. This is fine, since we are training a single overall `DoubleDQNAgent`, but it complicates the collection of the tuples described above. In order to carry out the training process in our setting, at each step we need to provide access to

1. the `(state, action, reward, next_state, done)` of the current `Agent`;
2. the `state` of the following `Agent` in the priority queue.

This will be easy to do, but may require adding onto the standard `gym` interface.

### Question
How should we go about this?

**Comment by DavidNKraemer** [comment]
This can be resolved by changing `self.state` to `agent.state` in line 96 of [src/envs.py](https://github.com/DavidNKraemer/SoCG21/blob/dev/src/envs.py).

**Comment by DavidNKraemer** [comment]
Implemented in commit f4949514e42d82b1a132290269f6e6771167c50e, closing issue.
